libs/community/langchain_community/chat_models/maritalk.py

```diff
@@ -163,6 +178,75 @@ def _call(
         # Fallback return statement, in case of unexpected code paths
         return "An unexpected error occurred"
 
+    async def _astream(
+        self,
+        messages: List[BaseMessage],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[AsyncCallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> AsyncIterator[ChatGenerationChunk]:
+        if "httpx" not in sys.modules:
+            import httpx
+
+        headers = {"Authorization": f"Key {self.api_key}"}
+        stopping_tokens = stop if stop is not None else []
+
+        parsed_messages = self.parse_messages_for_model(messages)
+
+        data = {
+            "messages": parsed_messages,
+            "model": self.model,
+            "do_sample": self.do_sample,
+            "max_tokens": self.max_tokens,
+            "temperature": self.temperature,
+            "top_p": self.top_p,
+            "stopping_tokens": stopping_tokens,
+            "stream": True,
+            **kwargs,
+        }
+
+        async with httpx.AsyncClient() as client:
+            async with client.stream(
+                "POST",
+                "https://chat.maritaca.ai/api/chat/inference",
+                data=json.dumps(data),
+                headers=headers,
+            ) as response:
+                response.raise_for_status()
+                async for line in response.aiter_lines():
+                    if line.startswith("data: "):
+                        data = line.replace("data: ", "")
+                        if data:
+                            parsed_data = json.loads(data)
+                            if "text" in parsed_data:
+                                delta = parsed_data["text"]
+                                chunk = ChatGenerationChunk(
+                                    message=AIMessageChunk(content=delta)
+                                )
+                                if run_manager:
+                                    await run_manager.on_llm_new_token(
+                                        delta, chunk=chunk
+                                    )
+                                yield chunk
+
+    async def _agenerate(
+        self,
+        messages: List[BaseMessage],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[AsyncCallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> ChatResult:
+        if self.streaming:
```

Comment:
@rodrigo-f-nogueira do you intend for `.streaming` to be an attribute of `ChatMaritalk`? if so you need to add it to the class definition.

Reply:
Hi @ccurme , this should be fixed now